src/Protocol/ActivityPub/FetchQueue.php

```diff
@@ -48,10 +50,17 @@ public function push(FetchQueueItem $item)
 	 */
 	public function process()
 	{
+		$failedIds = [];
+
 		while (count($this->queue)) {
 			$fetchQueueItem = array_pop($this->queue);
 
-			call_user_func_array([Processor::class, 'fetchMissingActivity'], array_merge([$this], $fetchQueueItem->toParameters()));
+			if (!call_user_func_array([Processor::class, 'fetchMissingActivity'], array_merge([$this], $fetchQueueItem->toParameters()))) {
+				$failedIds[] = $fetchQueueItem->getUrl();
+			}
 		}
+
+		// Removing orphans if fetch failed
+		Post::delete(['thr-parent' => $failedIds]);
```

Comment:
@annando Is there a more complete way of doing this that includes all related tables or is it taken care of with the cascade attribute? I'm thinking about the `uri-id` for example?